### PR TITLE
New typography

### PR DESCRIPTION
**Pull request by connor-baer**
Addresses [DSYS-708](https://sumupteam.atlassian.net/browse/DSYS-708) and [DSYS-711](https://sumupteam.atlassian.net/browse/DSYS-711).

## Purpose

One of the biggest challenges in our user interfaces today is how to convene a clear hierarchy of the elements presented on a screen. Text plays a critical role in tackling this challenge, and currently our typographic scale has several limitations to our developers' and designers' needs. This is particularly crucial for black-and-white interfaces, where the absence of color places greater emphasis on typography to convey hierarchy and guide the user's eye. To address this, we need to establish a clearer hierarchy through a more consistent and scalable typographic system, ensuring better readability and comprehension across different platforms.

## Approach and changes

- Adjust and simplify the typescale. The legacy font sizes and line heights are mapped to the new values so teams can benefit from this improvement without code changes. Teams should gradually migrate to the new size values which are aligned across platforms. This can be automated using the existing `no-renamed-props` and the new `no-deprecated-custom-properties` ESLint and Stylelint rules.
- Rename and consolidate the typography components for cross-platform consistency. The Title component has been renamed to Display, the SubHeadline component has been deprecated in favor of the Headline component in size `s`, and the BodyLarge component has been deprecated in favor of the Body component in size `l`. 
- Deprecate the Body component's `variant` prop in favor of the new `weight` and `color` props.
- Add a new Compact component for space-constraint use cases and add a new Numeral component to display numeric strings such as currency values at various sizes.

> [!WARNING]  
> The typography tokens and new component APIs are not final and subject to breaking changes during the prerelease phase.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Review by marcelvoss — APPROVED** [review]
Makes sense to me! Will have a look at the iOS implementation and see how I can apply the same there.

File: packages/circuit-ui/components/Display/Display.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+The Display component comes in three sizes. In most cases, use the [Headline component](Typography/Headline) component instead to render headings.
+
+<Story of={Stories.Sizes} />
+
+---
+
+## Accessibility
+
+All accessibility guidelines for the [Headline component](Typography/Headline) also apply to the Display component../Display.stories
```
Comment (connor-baer) [inline packages/circuit-ui/components/Display/Display.mdx:27]:
```suggestion
All accessibility guidelines for the [Headline component](Typography/Headline) also apply to the Display component.
```

File: docs/features/1-theme.mdx [hunk 2]
```diff
@@ -61,7 +61,9 @@ Use spacings for gutters, margins, and paddings. Don't use it for border width,
 
 ## Typography
 
-Avoid using the `var(--cui-typography-*)` CSS custom properties directly in your styles. Instead, use the typography components [`Title`](Typography/Title/Docs), [`Headline`](Typography/Headline/Docs), [`SubHeadline`](Typography/SubHeadline/Docs), [`Body`](Typography/Body/Docs), and [`BodyLarge`](Typography/BodyLarge/Docs).
+Avoid using the `var(--cui-typography-*)` CSS custom properties directly in your styles. Instead, use the typography components [`Display`](Typography/Display/Docs), [`Headline`](Typography/Headline/Docs), [`SubHeadline`](Typography/SubHeadline/Docs), [`Body`](Typography/Body/Docs), and [`BodyLarge`](Typography/BodyLarge/Docs).
```
Comment (connor-baer) [inline docs/features/1-theme.mdx:64]:
```suggestion
Avoid using the `var(--cui-typography-*)` CSS custom properties directly in your styles. Instead, use the typography components [`Display`](Typography/Display/Docs), [`Headline`](Typography/Headline/Docs), and [`Body`](Typography/Body/Docs).
```

File: packages/circuit-ui/components/Headline/Headline.mdx [hunk 3]
```diff
@@ -16,9 +16,9 @@ The Headline component is used for describing the contents of a page or page sec
 
 ### Sizes
 
-The Headline component comes in four sizes.
+The Headline component comes in three sizes.
 
-Use the `four` size for card headers and `three` for page titles in web applications. For specific use cases such as landing pages, consider using the [Title](Typography/Title) component.
+Use the `four` size for card headers and `three` for page titles in web applications. For specific use cases such as landing pages, consider using the [Display](Typography/Display) component.
```
Comment (connor-baer) [inline packages/circuit-ui/components/Headline/Headline.mdx:21]:
```suggestion
Use the `m` size for card headers and `l` for page titles in web applications. Consider using the [Display](Typography/Display) component for specific use cases such as landing pages.
```

File: packages/stylelint-plugin-circuit-ui/no-deprecated-custom-properties/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+n/a
+
+## When Not To Use It
+
+n/a
+
+## Further Reading
+
+- [Theme documentation](https://circuit.sumup.com/?path=/docs/features-theme--docs) on the Circuit UI docs
```
Comment (connor-baer) [inline packages/stylelint-plugin-circuit-ui/no-deprecated-custom-properties/README.md:29]:
```suggestion
- [Theme documentation](https://circuit.sumup.com/?path=/docs/features-theme--docs) on the Circuit UI docs
- [Migration guide](https://github.com/sumup-oss/circuit-ui/blob/main/MIGRATION.md)
- [Design token release notes](https://github.com/sumup-oss/circuit-ui/blob/main/packages/design-tokens/CHANGELOG.md)
```

File: packages/eslint-plugin-circuit-ui/no-deprecated-custom-properties/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+n/a
+
+## When Not To Use It
+
+n/a
+
+## Further Reading
+
+- [Migration guide](https://github.com/sumup-oss/circuit-ui/blob/main/MIGRATION.md)
```
Comment (connor-baer) [inline packages/eslint-plugin-circuit-ui/no-deprecated-custom-properties/README.md:35]:
```suggestion
- [Theme documentation](https://circuit.sumup.com/?path=/docs/features-theme--docs) on the Circuit UI docs
- [Migration guide](https://github.com/sumup-oss/circuit-ui/blob/main/MIGRATION.md)
```

File: packages/circuit-ui/components/Body/Body.mdx [hunk 6]
```diff
@@ -22,6 +22,12 @@ The Body component comes in three sizes. Use the default `m` size in most cases.
 
 <Story of={Stories.Sizes} />
 
+### Weights
+
+The Body component comes in three weights. Use the default `regular` weight in most cases.
```
Comment (sirineJ) [inline packages/circuit-ui/components/Body/Body.mdx]: `The Body component comes in three weights.`
but there is only 2 in the component 
`const weights = ['regular', 'bold'] as const;`
 ?

**Pull request merged**